jerry-core/ecma/operations/ecma-conversion.c

```diff
@@ -355,7 +355,19 @@ ecma_op_to_string (ecma_value_t value) /**< ecma value */
       res_p = ecma_get_string_from_value (value);
       res_p = ecma_copy_or_ref_ecma_string (res_p);
     }
-    else if (ecma_is_value_number (value))
+    else if (ecma_is_value_integer_number (value))
+    {
+      if ((ecma_integer_value_t) value < 0)
+      {
+        ecma_number_t num = ecma_get_number_from_value (value);
+        res_p = ecma_new_ecma_string_from_number (num);
+      }
+      else
+      {
+        res_p = ecma_new_ecma_string_from_uint32 ((uint32_t) (ecma_get_integer_from_value (value)));
+      }
+    }
+    else if (ecma_is_value_float_number (value))
```

Comment:
I would call ecma_get_integer_from_value (value) first, then check the sign, and that makes ecma_get_number_from_value call unnecessary.